include/dpp/snowflake.h

```diff
@@ -52,141 +53,204 @@ class DPP_EXPORT snowflake final {
 	/**
 	 * @brief The snowflake value
 	 */
-	uint64_t value;
+	uint64_t value = 0;
 
 public:
 	/**
 	 * @brief Construct a snowflake object
-	 * @param value A snowflake value
 	 */
-	snowflake(const uint64_t& value);
+	constexpr snowflake() noexcept = default;
 
 	/**
-	 * @brief Construct a snowflake object
+	 * @brief Copy a snowflake object
+	 */
+	constexpr snowflake(const snowflake &rhs) noexcept = default;
+
+	/**
+	 * @brief Move a snowflake object
+	 */
+	constexpr snowflake(snowflake &&rhs) noexcept = default;
+
+	/**
+	 * @brief Construct a snowflake from an integer value
+	 *
+	 * @throw dpp::logic_exception on assigning a negative value. the function is noexcept if the type given is unsigned
```

Comment:
Im gonna be that guy, you need a capital on "the", should be `... negative value. The function...`